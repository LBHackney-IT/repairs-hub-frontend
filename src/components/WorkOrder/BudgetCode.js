import { formatBudgetCode } from '@/utils/helpers/budgetCodes'

const BudgetCode = ({ budgetCode }) => {
  if (!budgetCode) {
    return null
  }

  return (
    <p className="govuk-body govuk-!-margin-bottom-0 lbh-!-font-weight-bold">
      Budget code â€“ Subjective:
      <br />
      {formatBudgetCode(budgetCode)}
    </p>
  )
}

export default BudgetCode
