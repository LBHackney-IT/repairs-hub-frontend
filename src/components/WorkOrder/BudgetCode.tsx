import { formatBudgetCode } from '@/utils/helpers/budgetCodes'
import { BudgetCode as BudgetCodeType } from '../../models/budgetCode'

interface Props {
  budgetCode: BudgetCodeType
}

const BudgetCode = ({ budgetCode }: Props) => {
  if (!budgetCode) return null

  return (
    <p className="govuk-body govuk-!-margin-bottom-0 lbh-!-font-weight-bold">
      Budget code â€“ Subjective:
      <br />
      {formatBudgetCode(budgetCode)}
    </p>
  )
}

export default BudgetCode
