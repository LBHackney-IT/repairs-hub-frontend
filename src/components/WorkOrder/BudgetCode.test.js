import { render } from '@testing-library/react'
import BudgetCode from './BudgetCode'

describe('BudgetCode component', () => {
  it('details the budget code', () => {
    const { asFragment } = render(
      <BudgetCode
        budgetCode={{
          externalCostCode: 'externalCostCode',
          corporateSubjectiveCode: 'corporateSubjectiveCode',
          descriptionOfWorks: 'descriptionOfWorks',
        }}
      />
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
