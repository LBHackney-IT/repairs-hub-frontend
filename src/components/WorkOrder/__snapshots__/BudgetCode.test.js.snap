// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BudgetCode component details the budget code 1`] = `
<DocumentFragment>
  <p
    class="govuk-body govuk-!-margin-bottom-0 lbh-!-font-weight-bold"
  >
    Budget code â€“ Subjective:
    <br />
    externalCostCode - corporateSubjectiveCode descriptionOfWorks
  </p>
</DocumentFragment>
`;
